cpg_infra/abstraction/group_settings.py

```diff
@@ -0,0 +1,113 @@
+"""
+Contains pulumi.dynamic.ResourceProvider implementations
+for Google Groups settings.
+"""
+
+from functools import cache
+import pulumi
+import pulumi.dynamic
+import google.auth
+import googleapiclient.discovery
+from google.auth.transport.requests import Request
+
+
+class GroupSettings(pulumi.dynamic.Resource):
+    """A Pulumi dynamic resource for Google Groups settings."""
+
+    group_email: pulumi.Output[str]
+    settings: pulumi.Output[dict]
```

Comment:
Is it worth defining the group settings as constants? Or could you direct link to the docs where those settings are defined?

Reply:
Good idea to add a link. Done!